File: ui/src/app/modules/funcionario/models/usuario.ts
import { Permissao } from '../../seguranca/models/permissao';

export class Usuario {
  public idUsuario: string;
  public senha: string;
  public permissoes: Array<Permissao>;

  public isValid(): boolean {
    return this.isIdUsuarioValid()
      && this.isSenhaValid();
  }

  private isIdUsuarioValid(): boolean {
    return this.isStringValid(this.idUsuario);
  }

  private isSenhaValid(): boolean {
    return this.isStringValid(this.senha);
  }

  private isStringValid(text: string): boolean {
    return (text && text.length !== 0);
  }
}

File: api/src/main/java/com/prep/api/model/usuario/Usuario.java
package com.prep.api.model.usuario;

import com.prep.api.model.ModelAbstract;
import com.prep.api.model.permissao.Permissao;

import java.util.Set;

public class Usuario extends ModelAbstract {

    protected String idUsuario;
    protected String senha;

    protected Set<? extends Permissao> permissoes;

    public String getIdUsuario() {
        return idUsuario;
    }

    public void setIdUsuario(String idUsuario) {
        this.idUsuario = idUsuario;
    }

    public String getSenha() {
        return senha;
    }

    public void setSenha(String senha) {
        this.senha = senha;
    }
}
